dipy/tracking/local/localtracking.py

```diff
@@ -68,66 +71,187 @@ def __init__(self, direction_getter, tissue_classifier, seeds, affine,
+            Used to get directions for fiber tracking.
+        tissue_classifier : instance of ConstrainedTissueClassifier
+            Identifies endpoints and invalid points to inform tracking.
+        seeds : array (N, 3)
+            Points to seed the tracking. Seed points should be given in point
+            space of the track (see ``affine``).
+        affine : array (4, 4)
+            Coordinate space for the streamline point with respect to voxel
+            indices of input data. This affine can contain scaling, rotational,
+            and translational components but should not contain any shearing.
+            An identity matrix can be used to generate streamlines in "voxel
+            coordinates" as long as isotropic voxels were used to acquire the
+            data.
+        step_size : float
+            Step size used for tracking.
+        max_cross : int or None
+            The maximum number of direction to track from each seed in crossing
+            voxels. By default all initial directions are tracked.
+        maxlen : int
+            Maximum number of steps to track from seed. Used to prevent
+            infinite loops.
+        pft_back_tracking_dist : float
+            Distance in mm to back track before starting the particle filtering
+            tractography. The total particle filtering tractography distance is
+            equal to back_tracking_dist + front_tracking_dist.
+            By default this is set to 2 mm.
+        pft_front_tracking_dist : float
+            Distance in mm to run the particle filtering tractography after the
+            the back track distance. The total particle filtering tractography
+            distance is equal to back_tracking_dist + front_tracking_dist. By
+            default this is set to 1 mm.
+        pft_max_trial : int
+            Maximum number of trial for the particle filtering tractography
+            (Prevents infinite loops).
+        particle_count : int
+            Number of particles to use in the particle filter.
+        return_all : bool
+            If true, return all generated streamlines, otherwise only
+            streamlines reaching end points or exiting the image.
+        """
+
+        if not isinstance(tissue_classifier, ConstrainedTissueClassifier):
+            raise ValueError("expecting ConstrainedTissueClassifier")
+
+        self.pft_max_nbr_back_steps = int(np.ceil(pft_back_tracking_dist
+                                                  / step_size))
+        self.pft_max_nbr_front_steps = int(np.ceil(pft_front_tracking_dist
+                                                   / step_size))
+        pft_max_steps = (self.pft_max_nbr_back_steps +
+                         self.pft_max_nbr_front_steps)
+
+        if (self.pft_max_nbr_front_steps < 0
+                or self.pft_max_nbr_back_steps < 0
+                or pft_max_steps < 1):
+            raise ValueError("The number of PFT steps must be greater than 0.")
```

Comment:
Wouldn't `<= 0.0` be better for `pft_max_nbr_front_steps` and `pft_max_nbr_back_steps`? It seems to me that `pft_back_tracking_dist` and `pft_front_tracking_dist` should be > 0.0. No?

Reply:
At least one of the 2 must be `>0`, otherwise, there is no PFT.  Putting `pft_max_nbr_front_steps = 0` is conceivable, with a `pft_back_tracking_dist > 0`, or the other way around. 

With `CMC` (probabilistic stopping criteria), `pft_back_tracking_dist` can be 0 without issue, that wouldn't work for `ACT` (deterministic stopping criteria), i.e. with `ACT` and `pft_back_tracking_dist = 0`, the output would be the same as without PFT.